### ISSUE TITLE
Deck Calendars don't show on mobile

### ISSUE THREAD

**Issue by jospoortvliet** [nextcloud/deck#3103]
I have a few Deck calendars with deadlines in the Calendar app, but I don't see them on my Android mobile in DavX5 as optional calendars (I do see a lot of others). After checking, Frank and Vanessa don't see them on their iOS devices but Arthur claims his Sailfish does show them. Magic!

I'm not sure if this is intentional or not but - here's an issue so it can, potentially, be tracked and addressed.

**Comment by tcitworld** [comment]
Oh, it could be that Deck exports `VTODO`s and you might not see them in your calendar app. Try with OpenTasks?

https://github.com/nextcloud/deck/issues/2345#issuecomment-703455027

**Comment by mstyp** [comment]
> Oh, it could be that Deck exports `VTODO`s and you might not see them in your calendar app. Try with OpenTasks?
> 
> [#2345 (comment)](https://github.com/nextcloud/deck/issues/2345#issuecomment-703455027)

The problem is that there are many calendar systems that don't support `VTODOs`, like outlook or google. Which makes it impossible to use deck in an interoperably.

**Comment by makemeunsee** [comment]
I'm suffering from the same issue, trying to have the deck calendars shown on mobile via DAVx5.
The calendars show up as todo lists and my calendar app apparently doesnt know what how to treat those.

Would it be just about changing https://github.com/nextcloud/deck/blob/master/lib/DAV/Calendar.php#L208?

**Comment by tcitworld** [comment]
> Would it be just about changing https://github.com/nextcloud/deck/blob/master/lib/DAV/Calendar.php#L208?

Even though this would allow the CalDAV collection to hold `VEVENT`s, the issue the issue is that deck cards [are serialized as TODOs](https://github.com/nextcloud/deck/blob/master/lib/Db/Card.php#L135-L165), which is necessary to have completed/progress features.

Android only allows to expose events to calendars, not tasks, so the only way is to use a task app like OpenTasks or tasks.org. See https://www.davx5.com/faq/tasks/advanced-task-features and the associated picture.

**Comment by makemeunsee** [comment]
yes, I've just had a further look and realized changing the card type to `VEVENT` wouldnt make sense.

> Android only allows to expose events to calendars
that's a pity, thanks for the information, that spares me some wasted effort...

**Comment by tcitworld** [comment]
As a summary:
- Deck converts cards to Tasks (icalendar `VTODO`) so that we have due date and completed status. Regular events (`VEVENT`) cannot hold such information so they can't be use instead.
- The Android Calendar storage (a build-in Android thing that all calendar apps use) doesn't support showing tasks¹.
- The DAVx⁵ supports syncing tasks to other apps such as OpenTasks, jtx Board and Tasks.org (see figure below), and these apps should display the tasks corresponding to cards properly (in read-only).
- Non-android systems should work properly (iOS & macOS have build-in support for icalendar & CalDAV)

Since there's not much we can do on Nextcloud's side, I suggest to close this issue.

¹ A notable exception seems to be aCalendar [which says it can use tasks from OpenTasks if installed](https://acalendar.tapirapps.de/en/support/solutions/articles/36000058826-tasks-feature), but it seems to be a separate module, not sure if tasks are shown directly into the calendar view (is that correct @dodedodo ?).

![](https://www.davx5.com/fileadmin/user_upload/faq/how_davx5_works.svg)
